content/local-setup.md

```diff
@@ -0,0 +1,283 @@
+
+```bash
+solana program deploy <PATH>
+```
+
+Lastly, you can view the details for Solana CLI commands using the `solana --help` command.
+
+```bash
+solana --help
+```
+
+## Solana Program from Local Environment
+
+To create a new Rust project run the following in a new terminal.
+
+```bash
+cargo new --lib <PROJECT_FILE_NAME>
+```
+
+Open the new project and update the `Cargo.toml` file. Remember to specify the latest version of `solana-program`.
+
+```rust
+[package]
+name = "<PROJECT_FILE_NAME>"
+version = "0.1.0"
+edition = "2021"
+
+[features]
+no-entrypoint = []
+
+[dependencies]
+solana-program = "1.10.7"
+
+[lib]
+crate-type = ["cdylib", "lib"]
+```
+
+To build a Solana program in Rust, run the `cargo build-bpf` command. The output provides the `solana program deploy` command specifying the file path to deploy the program.
+
+```bash
+cargo build-bpf
+```
+
+Finally, to deploy the program after building, run the `solana program deploy` command output from `cargo build-bpf`.
+
+```bash
+solana program deploy <PATH>
+```
+
+# Demo
+
+Let’s practice by building and deploying the “Hello World!” program from lesson one of this module.
+
+### 1. Setup Solana Config
+
+First, set to endpoint to Devnet using the `solana config set --url` ****command.
+
+```bash
+solana config set --url https://api.devnet.solana.com
+```
+
+Next, check that the Solana CLI configuration has updated using the `solana config get` command.
+
+```bash
+solana config get
+```
+
+### 2. Create New Rust Project
+
+Now we are ready to create our a new program.  Run the `cargo new --lib` command.
+
+```bash
+cargo new --lib <PROJECT_FILE_NAME>
+```
+
+Update `cargo.toml` with the dependencies we’ll need to build a Solana program.
+
+```bash
+[package]
+name = "<PROJECT_FILE_NAME>"
+version = "0.1.0"
+edition = "2021"
+
+[features]
+no-entrypoint = []
+
+[dependencies]
+solana-program = "1.10.7"
+
+[lib]
+crate-type = ["cdylib", "lib"]
+```
+
+### 3. Build and Deploy Program
+
+Next, update `lib.rs` with the “Hello World!” program below. This program simply prints “Hello, world!” to the program log when the program is invoked.
+
+```rust
+use solana_program::{
+  account_info::AccountInfo,
+  entrypoint,
+  entrypoint::ProgramResult,
+  pubkey::Pubkey,
+  msg
+};
+
+entrypoint!(process_instruction);
+
+pub fn process_instruction(
+  program_id: &Pubkey,
+  accounts: &[AccountInfo],
+  instruction_data: &[u8]
+) -> ProgramResult{
+  msg!("Hello, world!");
+
+  Ok(())
+}
+```
+
+Next, build the program using by running the `cargo build-bpf` command.
+
+```bash
+cargo build-bpf
+```
+
+Finally, deploy the program using the `solana program deploy` command output from `cargo build-bpf`.
+
+```bash
+solana program deploy <PATH>
+```
+
+The `solana program deploy` will output the `Program Id` for your program. You can now look up deployed program on Solana Explorer.
+
+# Challenge
```

Comment:
I think we need to modify the challenge. They should be doing what they're doing in this challenge in the demo, and then the challenge they should be doing all of it all over again independently.

Reply:
update challenge to build and deploy on devnet